core/tauri-utils/src/config.rs

```diff
@@ -1038,6 +1033,33 @@ impl BundleResources {
   }
 }
 
+/// Updater type
+#[derive(Debug, PartialEq, Eq, Clone, Deserialize, Serialize)]
+#[cfg_attr(feature = "schema", derive(JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields, untagged)]
+pub enum Updater {
+  /// Generates lagacy zipped v1 compatible updaters
+  String(V1Compatible),
+  /// Produce updaters and their signatures or not
+  // Can't use untagged on enum field here: https://github.com/GREsau/schemars/issues/222
+  Bool(bool),
+}
+
+impl Default for Updater {
+  fn default() -> Self {
+    Self::Bool(false)
```

Comment:
@lucasfernog should we make `V1Compatible` the default and emit a warning that it will be changed in v3?

Reply:
Enable generating updater only make sense if they use the updater plugin, I don't enable should be the default (or maybe we detect plugin updater field), and `V1Compatible` should never be the default, I'm so worried about we will end up having everyone still using `V1Compatible`, and warning on default value is pretty weird as well

Maybe we change the name to `LegacyV1Compatible` so that people will know it's not supposed to be used (maybe some more details in the description), and should really only be there because of `tauri migrate`

Reply:
The problem is in order for apps built with v1, to migrate to updater v2, they need to push zips so v1 can update v2, then they can stop using them

Reply:
that's what I am suggesting, keep the current default and constantly emit warnings that it will be removed in v3. This allows users to migrate their v1 apps to v2 and then later from v2 to v3 there won't be an issue.

Reply:
kk, I can add that warning tomorrow